homeassistant/components/fish_audio/config_flow.py

```diff
@@ -0,0 +1,390 @@
+    )
+
+
+def get_filter_schema(options: TTSConfigData) -> vol.Schema:
+    """Return the schema for the filter step."""
+    return vol.Schema(
+        {
+            vol.Optional(
+                CONF_SELF_ONLY, default=options.get(CONF_SELF_ONLY, False)
+            ): bool,
+            vol.Optional(
+                CONF_LANGUAGE, default=options.get(CONF_LANGUAGE, "en")
+            ): LanguageSelector(
+                LanguageSelectorConfig(
+                    languages=TTS_SUPPORTED_LANGUAGES,
+                )
+            ),
+            vol.Optional(
+                CONF_SORT_BY, default=options.get(CONF_SORT_BY, "score")
+            ): SelectSelector(
+                SelectSelectorConfig(
+                    options=SORT_BY_OPTIONS, mode=SelectSelectorMode.DROPDOWN
+                )
+            ),
+        }
+    )
+
+
+def get_model_selection_schema(
+    options: TTSConfigData, model_options: list[SelectOptionDict]
+) -> vol.Schema:
+    """Return the schema for the model selection step."""
+    return vol.Schema(
+        {
+            vol.Required(
+                CONF_VOICE_ID,
+                default=options.get(CONF_VOICE_ID, ""),
+            ): SelectSelector(
+                SelectSelectorConfig(
+                    options=model_options,
+                    mode=SelectSelectorMode.DROPDOWN,
+                    custom_value=True,
+                )
+            ),
+            vol.Required(
+                CONF_BACKEND,
+                default=options.get(CONF_BACKEND, "s1"),
+            ): SelectSelector(
+                SelectSelectorConfig(
+                    options=[
+                        SelectOptionDict(value=opt, label=opt) for opt in BACKEND_MODELS
+                    ],
+                    mode=SelectSelectorMode.DROPDOWN,
+                )
+            ),
+        }
+    )
+
+
+def get_name_schema(options: TTSConfigData, default: str | None = None) -> vol.Schema:
+    """Return the schema for the name input."""
+    return vol.Schema({vol.Required(CONF_NAME, default=default or vol.UNDEFINED): str})
+
+
+async def _validate_api_key(
+    hass: HomeAssistant, api_key: str
+) -> tuple[APICreditEntity, Session]:
+    """Validate the user input allows us to connect."""
+    session = Session(api_key)
+
+    try:
+        credit_info = await hass.async_add_executor_job(session.get_api_credit)
+    except HttpCodeErr as exc:
+        if exc.status == 401:
+            raise InvalidAuthError(exc) from exc
+        raise CannotConnectError(exc) from exc
+    except Exception as exc:
+        raise UnexpectedError(exc) from exc
+
+    return credit_info, session
+
+
+class FishAudioConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Fish Audio."""
+
+    VERSION = 1
+    _reauth_entry: ConfigEntry | None
+    session: Session | None
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._reauth_entry = None
```

Comment:
1. Let's split reauth off for now
2. you can get that via a method on the superclass